src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java

```diff
@@ -0,0 +1,490 @@
+              conn.execute("INSERT INTO test_nulls_table (id, name) VALUES (1, NULL)", ar4 -> {
+                ensureSuccess(context, ar4);
+                conn.query("SELECT id, name FROM test_nulls_table ORDER BY id", ar5 -> {
+                  ensureSuccess(context, ar5);
+                  ResultSet rs = ar5.result();
+
+                  List<String> columns = rs.getColumnNames();
+                  context.assertEquals(2, columns.size());
+                  context.assertEquals(columns.get(0), "id");
+                  context.assertEquals(columns.get(1), "name");
+
+                  context.assertEquals(rs.getResults().size(), 1);
+                  context.assertEquals(rs.getResults().get(0).getInteger(0), 1);
+                  context.assertEquals(rs.getResults().get(0).getString(1), null);
+                  async.complete();
+                });
+              });
+            });
+      });
+    });
+  }
+
+  public static final String insertedTime1 = "2015-02-22T07:15:01.234Z";
+
+  public static final String expectedTime1 = "2015-02-22T07:15:01.";
+
+  public static final String insertedTime2 = "2014-06-27T17:50:02.468+02:00";
+
+  public static final String expectedTime2 = "2014-06-27T17:50:02.";
+
+
+  @Test
+  public void testDateValueSelection(TestContext context) {
+    Async async = context.async();
+    client.getConnection(ar -> {
+      ensureSuccess(context, ar);
+      conn = ar.result();
+      conn.execute("DROP TABLE IF EXISTS test_date_table", ar2 -> {
+        ensureSuccess(context, ar2);
+        conn.execute("CREATE TABLE test_date_table (id BIGINT, some_date DATE,some_timestamp TIMESTAMP)", ar3 -> {
+          ensureSuccess(context, ar3);
+          conn.updateWithParams("INSERT INTO test_date_table (id, some_date, some_timestamp) VALUES (?, ?, ?)", new JsonArray().add(1).add("2015-02-22").add(insertedTime1), ar4 -> {
+            ensureSuccess(context, ar4);
+            conn.updateWithParams("INSERT INTO test_date_table (id, some_date, some_timestamp) VALUES (?, ?, ?)", new JsonArray().add(2).add("2007-07-20").add(insertedTime2), ar5 -> {
+              ensureSuccess(context, ar5);
+              conn.query("SELECT id, some_date, some_timestamp FROM test_date_table ORDER BY id", ar6 -> {
+                ensureSuccess(context, ar6);
+                ResultSet results = ar6.result();
+                List<String> columns = results.getColumnNames();
+                context.assertEquals(3, columns.size());
+                context.assertEquals("id", columns.get(0));
+                context.assertEquals("some_date", columns.get(1));
+                context.assertEquals("some_timestamp", columns.get(2));
+
+                context.assertEquals(2, results.getResults().size());
+                JsonArray row1 = results.getResults().get(0);
+                context.assertEquals(row1.getString(1), "2015-02-22");
+                context.assertTrue(row1.getString(2).startsWith(expectedTime1));
```

Comment:
I had different expectations for MySQL and PostgreSQL by subclassing and changing the `expectedTime` variables for each version (MySQL = .000 and PostgreSQL = .234). Not sure if just checking for `startsWith` is the best way to deal with this...